src/dissect/dissect.c

```diff
@@ -225,7 +237,7 @@ static int run(int argc, char *argv[]) {
 
                         k = PARTITION_VERITY_OF(i);
                         if (k >= 0)
-                                printf(" %s verity", m->partitions[k].found ? "with" : "without");
+                                printf(" %s verity", m->partitions[k].found || m->verity ? "with" : "without");
```

Comment:
given the test is a bit more complex now: could you please add a helper call dissected_image_has_verity() in dissect-image.c or so that returns a simple boolean  and internalizes the test? That would make things a lot more readable

Reply:
Sure, added 2 helpers - _has_verity and _can_do_verity